radio_beam/beam.py

```diff
@@ -245,29 +267,12 @@ def deconvolve(self, other):
         limit = np.min(axes)
         limit = 0.1*limit*limit
         
-        # two cases...
-
-        # ... failure
         if (alpha < 0) or (beta < 0) or (s < t):
-
-            # Note failure as an empty beam
-            new_major = 0.0
-            new_minor = 0.0
-            new_pa = 0.0
-
-            # Record that things failed
-            failed = True
-            
-            # Check if it is close to a point source
-            if ((0.5*(s-t) < limit) and
-                (alpha > -1*limit) and
-                (beta > -1*limit)):
-                pointlike = True
+            if failure_returns_pointlike:
```

Comment:
Just as importantly, though - I don't think this represents an actual change in functionality, except that one can optionally specify that all failures should be assumed pointlike.  Since the default is `False`, I think we're in OK territory - it should take real effort to get unexpected results.